src/commands/revisionDraft/createRevisionDraft.ts

```diff
@@ -0,0 +1,131 @@
+import { pickRevisionItem } from "../../utils/pickRevisionItem";
+import { IContainerAppContext } from "../IContainerAppContext";
+
+export async function createRevisionDraft(context: IActionContext, node?: RevisionItem | RevisionsItem): Promise<void> {
+    const containerAppsItem = node ?? await pickContainerApp(context);
+
+    if (ext.revisionDraftFileSystem.hasRevisionDraft(containerAppsItem)) {
+        throw new Error(localize('revisionDraftExists', 'A revision draft already exists for container app "{0}".', containerAppsItem.containerApp.name));
+    }
+
+    let revisionItem: RevisionItem;
+    if (containerAppsItem instanceof RevisionItem) {
+        revisionItem = containerAppsItem;
+    } else {
+        const { subscription, containerApp } = containerAppsItem;
+        const containerAppContext: IContainerAppContext = {
+            ...context,
+            ...createSubscriptionContext(subscription),
+            containerApp,
+            subscription
+        };
+
+        /**
+         * Overwrite the typical pick prompt behavior with custom behavior.
+         * Leverage the `selectByRevisionName` option to obtain the RevisionItem without re-prompting the user
+         */
+        const revisionName = await promptForRevisionName(containerAppContext);
+        revisionItem = await pickRevisionItem(context, containerAppsItem, {
+            selectByRevisionName: revisionName
+        });
+    }
+
+    let parentId: string;
+    if (revisionItem.containerApp.revisionsMode === KnownActiveRevisionsMode.Single) {
+        parentId = revisionItem.containerApp.id;
+    } else {
+        parentId = `${revisionItem.containerApp.id}/${RevisionsItem.idSuffix}`;
+    }
+
+    await ext.state.showCreatingChild(
+        parentId,
+        localize('creatingDraft', 'Creating draft...'),
+        async () => await ext.revisionDraftFileSystem.createOrEditRevisionDraft(revisionItem)
+    );
+}
+
+async function promptForRevisionName(context: IContainerAppContext): Promise<string | undefined> {
+    const revisionPicks: IAzureQuickPickItem<Revision | undefined>[] = await getRevisionNamePicks(context);
+    if (revisionPicks.length === 1) {
+        return revisionPicks[0].data?.name;
+    }
+
+    return (await context.ui.showQuickPick(revisionPicks, {
+        placeHolder: localize('selectBaseRevision', 'Select a base revision'),
+        suppressPersistence: true
+    })).data?.name;
+}
+
+dayjs.extend(relativeTime);
+
+async function getRevisionNamePicks(context: IContainerAppContext): Promise<IAzureQuickPickItem<Revision | undefined>[]> {
+    const rgName: string = nonNullValueAndProp(context.containerApp, 'resourceGroup');
+    const caName: string = nonNullValueAndProp(context.containerApp, 'name');
+
+    const client: ContainerAppsAPIClient = await createContainerAppsAPIClient(context);
+
+    const containerApp: ContainerApp = await client.containerApps.get(rgName, caName);
```

Comment:
Note to self - this line probably shouldn't be here... I think we can already get all the container app data we need off the context already